UKHO.FleetManagerMock.API/UKHO.FleetManagerMock.API/Controllers/FleetManagerController.cs

```diff
@@ -136,6 +136,119 @@ public IActionResult GetCatalogue([FromHeader(Name = "token")] string? token, [F
+                }
+            }
+            if (subscriptionKey == fleetManagerStubSubscriptionKey)
+            {
+                if (userPass == base64Credentials)
+                {
+                    string? authTokenResponse = "{\"token\": \"eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9\", \"expiration\":\"" + DateTime.UtcNow.ToString("yyyy-MM-ddTHH:mm:ssZ", CultureInfo.InvariantCulture) + "\"}";
+                    httpResponse.StatusCode = HttpStatusCode.OK;
+                    if (httpResponse.IsSuccessStatusCode)
+                    {
+                        AuthToken = authTokenResponse;
+                    }
+                }
+                else
+                {
+                    string? invalidUsernamePasswordResponse = "{\"correlationId\":\"10ce5b89-67f4-4090-8364-7ba69eb7cb3d\",\"errors\":[{\"source\":\"userPass\",\"description\":\"Missing or invalid encoded User:Pass.\"}]}";
+                    httpResponse.StatusCode = HttpStatusCode.BadRequest;
+                    if (!httpResponse.IsSuccessStatusCode)
+                    {
+                        return BadRequest(JsonConvert.DeserializeObject<InvalidUsernamePasswordResponse>(invalidUsernamePasswordResponse));
+                    }
+                }
+            }
+            else
+            {
+                string? invalidSubscriptionKeyResponse = "{\"statusCode\": 401, \"message\":\"Access denied due to invalid subscription key. Make sure to provide a valid key for an active subscription.\"}";
+                httpResponse.StatusCode = HttpStatusCode.Unauthorized;
+                if (!httpResponse.IsSuccessStatusCode)
+                {
+                    return Unauthorized(JsonConvert.DeserializeObject<InvalidSubscriptionKeyResponse>(invalidSubscriptionKeyResponse));
+                }
+            }
+            return Ok(JsonConvert.DeserializeObject<JwtAuthUnpToken>(AuthToken));
+        }
+
+        [HttpGet]
+        [Route("/ft/catalogues/{catalogueId}")]
+        public IActionResult GetCatalogueForFT([FromHeader(Name = "token")] string? token, [FromHeader(Name = "Ocp-Apim-Subscription-Key")] string? subscriptionKey)
+        {
+            Dictionary<string, string> requestHeaders = new();
+            foreach (KeyValuePair<string, StringValues> header in Request.Headers)
+            {
+                requestHeaders.Add(header.Key, header.Value);
+            }
+            string? fleetManagerStubSubscriptionKey = _fleetManagerApiConfiguration.Value.SubscriptionKey;
+            string? path = _fileDirectoryPathConfiguration.Value.FunctionalTestAVCSCatalogDataFilePath;
+            HttpResponseMessage httpResponse = new();
+
+            if (string.IsNullOrEmpty(subscriptionKey))
+            {
+                string? invalidSubscriptionKeyResponse = "{\"statusCode\": 401, \"message\":\"Access denied due to missing subscription key. Make sure to include subscription key when making requests to an API.\"}";
```

Comment:
invalidSubscriptionKeyResponse doesn't need to be nullable, and can be made into a const.

Reply:
Changed